bindings/csharp/Makefile

```diff
@@ -1,54 +1,54 @@
-INCLUDE_DIRS = ../../src ../../blst/bindings
-
-FIELD_ELEMENTS_PER_BLOB ?= 4096
-
 ifeq ($(OS),Windows_NT)
-	BLST_BUILDSCRIPT=./build.bat
-	BLST_OBJ=blst.lib
-	TESTS_EXECUTABLE=test_ckzg.exe
-	CSHARP_PLATFORM?=win-x64
-	CLANG_PLATFROM?=x86_64-win
-	CLANG_FLAGS=
-	CLANG_EXECUTABLE=clang
-	CKZG_LIBRARY_PATH=Ckzg.Bindings\runtimes\$(CSHARP_PLATFORM)\native\ckzg.dll
+    BLST_BUILDSCRIPT = ./build.bat
+    BLST_OBJ = blst.lib
+    CSHARP_PLATFORM ?= win-x64
+    CLANG_EXECUTABLE = clang
+    CKZG_LIBRARY_PATH = Ckzg.Bindings\runtimes\$(CSHARP_PLATFORM)\native\ckzg.dll
 else
-	BLST_BUILDSCRIPT=./build.sh
-	BLST_OBJ=libblst.a
-	TESTS_EXECUTABLE=./test_ckzg
-	CSHARP_PLATFORM?=linux-x64
-	CLANG_PLATFORM?=x86_64-linux
-	CLANG_FLAGS=-fPIC
-	CLANG_EXECUTABLE=cc
-	CKZG_LIBRARY_PATH=Ckzg.Bindings/runtimes/$(CSHARP_PLATFORM)/native/ckzg.so
+    BLST_BUILDSCRIPT = ./build.sh
+    BLST_OBJ = libblst.a
+    CLANG_EXECUTABLE = clang
+
+    UNAME_S := $(shell uname -s)
+    UNAME_M := $(shell uname -m)
+    ifeq ($(UNAME_S),Linux)
+        ifeq ($(UNAME_M),x86_64)
+            CSHARP_PLATFORM ?= linux-x64
+        else
+            CSHARP_PLATFORM ?= linux-arm64
+        endif
+    endif
+    ifeq ($(UNAME_S),Darwin)
+        ifeq ($(UNAME_M),arm64)
+            CSHARP_PLATFORM ?= osx-arm64
```

Comment:
Does it really work on arm macos? I'm not sure that github provides arm macos agent at all.

Reply:
Yes, that's what platform I made these changes on. Also, I believe it does. It's in the CI here:

https://github.com/ethereum/c-kzg-4844/blob/5c33098d6f401899f60c8cc7a6bafcba959188f6/.github/workflows/csharp-bindings-build.yml#L22-L27

Reply:
It's a clang target not a host os, so you can make arm build on x64 mac (macos-latest) setting target as I do here in my draft PR https://github.com/ethereum/c-kzg-4844/pull/46/files#diff-6fc44ed49a172f2e0114b9eac31616d29aac3c679e88877b9b675b884e1b8364R4 here
The current binding does not support macos arm actually. That's is why there is CLANG_PLATFORM, it should be used to specify the target.

Reply:
I'll add those changes after your PR

Reply:
Ah I understand. Okay thanks 👍